File: src/renderer/utils/defaultConfig.js
/**
 * @module count文件时的默认注释规则、默认过滤忽视的文件(夹)和默认二进制文件的类型
 * 
 **/

const Rule = {
    js: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ],
    go: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ],
    jsx: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ],
    vue: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        },
        {
            block: { start: '<!--', end: '-->' }
        }
    ],
    html: [
        {
            block: { start: '<!--', end: '-->' }
        }
    ],
    css: [
        {
            block: { start: '/*', end: '*/' }
        }
    ],
    java: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ],
    c: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ],
    cpp: [
        {
            block: { start: '/*', end: '*/' },
            row: '//'
        }
    ]
}

const ExcludePath = [
    'node_modules',
    '.git',
    '.gitignore',
    '.npmignore'
]

const BinaryType = ['jpg', 'png', 'jpeg', 'exe', 'pptx', 'ppt', 'docx', 'doc', 'xlsx', 'xls', 'pdf']

module.exports = {
    Rule,
    ExcludePath,
    BinaryType
}

File: count/defaultConfig.go
package count

// 块注释
type blockComment struct {
	start string
	end   string
}

// 注释规则
type CommentRule struct {
	row   string
	block blockComment
}

// 新建文件注释规则
func NewCommentRule(row, blockStart, blockEnd string) CommentRule {
	return CommentRule{
		row:   row,
		block: blockComment{blockStart, blockEnd},
	}
}

// 默认的文件注释规则map
var DefaultRuleConfig = make(map[string][]CommentRule)

// 默认过滤的文件(目录)slice
var ExcludePaths = []string{
	"node_modules",
	".git",
	".gitignore",
	".npmignore",
}

// 初始化文件的默认注释规则 DefaultRuleConfig
func init() {
	DefaultRuleConfig["js"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["ts"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["go"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["jsx"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["tsx"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["vue"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
		{
			block: blockComment{start: "<!--", end: "-->"},
		},
	}

	DefaultRuleConfig["svelte"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
		{
			block: blockComment{start: "<!--", end: "-->"},
		},
	}

	DefaultRuleConfig["html"] = []CommentRule{
		{
			block: blockComment{start: "<!--", end: "-->"},
		},
	}

	DefaultRuleConfig["css"] = []CommentRule{
		{
			block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["java"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["c"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}

	DefaultRuleConfig["cpp"] = []CommentRule{
		{
			row: "//", block: blockComment{start: "/*", end: "*/"},
		},
	}
}
